server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -2230,16 +2230,6 @@ public Closeable acquireHistoryRetentionLock() {
         return getEngine().acquireHistoryRetentionLock();
     }
 
-    /**
-     *
-     * Creates a new history snapshot for reading operations since
-     * the provided starting seqno (inclusive) and ending seqno (inclusive)
-     * The returned snapshot can be retrieved from either Lucene index or translog files.
-     */
-    public Translog.Snapshot getHistoryOperations(String reason, long startingSeqNo, long endSeqNo) throws IOException {
-        return getEngine().newChangesSnapshot(reason, startingSeqNo, endSeqNo, true);
```

Comment:
with this removed do we ever have a use case where newChangesSnapshot is called without requiredFullRange set to false and accurateCount set to true?

Reply:
Ohh. That's a good point. I need to look and see if CCR is using this method. I only removed it because it wasn't being used but we may need to keep it around.

Reply:
to avoid removing (for now) I added the method back w/ an optional `accurateCount` boolean. It doesn't look like this is used by CCR but I'm going to keep it until we deem it no longer useful